pkg/storage/cunofs.go

```diff
@@ -0,0 +1,146 @@
+package storage
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"text/template"
+	"time"
+
+	"github.com/beam-cloud/beta9/pkg/types"
+	"github.com/cenkalti/backoff"
+)
+
+const cunoFsMountTimeout time.Duration = 30 * time.Second
+
+type CunoFsStorage struct {
+	mountCmd *exec.Cmd
+	config   types.CunoFSConfig
+}
+
+func NewCunoFsStorage(config types.CunoFSConfig) (Storage, error) {
+	return &CunoFsStorage{
+		config: config,
+	}, nil
+}
+
+func (s *CunoFsStorage) Mount(localPath string) error {
+	log.Printf("CunoFS filesystem mounting to: '%s'\n", localPath)
+	s.mountCmd = exec.Command(
+		"cuno",
+		"mount",
+		localPath,
+		"--root",
+		fmt.Sprintf("s3://%s", s.config.S3BucketName),
+		"--mkdir",
+	)
+
+	// Start the mount command in the background
+	go func() {
+		output, err := s.mountCmd.CombinedOutput()
+		if err != nil {
+			log.Fatalf("error executing cunofs mount: %v, output: %s", err, string(output))
+		}
+	}()
+
+	ticker := time.NewTicker(100 * time.Millisecond)
+	timeout := time.After(cunoFsMountTimeout)
+
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case <-timeout:
+				done <- false
+				return
+			case <-ticker.C:
+				if isMounted(localPath) {
+					done <- true
+					return
+				}
+			}
+		}
+	}()
+
+	// Wait for confirmation or timeout
+	if !<-done {
+		return fmt.Errorf("failed to mount CunoFS filesystem to: '%s'", localPath)
+	}
+
+	log.Printf("CunoFS filesystem mounted to: '%s'", localPath)
+	return nil
+}
+
+func (s *CunoFsStorage) Format(fsName string) error {
+	activateCmd := exec.Command("sh", "-c", fmt.Sprintf("yes | echo %s | cuno creds activate", s.config.LicenseKey))
+	_, err := activateCmd.CombinedOutput()
+	if err != nil {
+		return err
+	}
+
+	templateStr := `aws_access_key_id={{.S3AccessKey}}
```

Comment:
This seems required. Is cuno able to use the injected aws credentials via the service account?

Reply:
Hm, looks like we'll use tigris for this anyway. Guess we can follow up later with using aws iam injected secrets.